### PR TITLE
Store the GUI always using the controller UUID.

### PR DESCRIPTION
**Pull request by frankban**
The GUI is always retrieved from the storage using the controller UUID.

### CODE REVIEW

File: state/binarystorage.go [hunk 1]
```diff
@@ -18,18 +18,16 @@ var binarystorageNew = binarystorage.New
 //
 // TODO(axw) remove this, add a constructor function in binarystorage.
 func (st *State) ToolsStorage() (binarystorage.StorageCloser, error) {
-	return newStorage(st, toolsmetadataC), nil
+	return newStorage(st.session.Copy(), st.ModelUUID(), toolsmetadataC), nil
```
Comment (rogpeppe) [inline state/binarystorage.go]: I think you probably want Clone here not Copy.

Comment (rogpeppe) [inline state/binarystorage.go]: ... but I think I'd probably just pass in the state as before and just add the uuid parameter, leaving the newStorage function responsible for cloning (or copying) the underlying session from within the state.

**Comment by rogpeppe** [comment]
LGTM with a couple of suggestions above, thanks.

File: state/binarystorage.go (hunk 1 above)
Comment (frankban) [inline state/binarystorage.go]: Done.

**Comment by alesstimec** [comment]
LGTM

**Comment by frankban** [comment]
$$merge$$

**Pull request merged**